optimum/exporters/executorch/integrations.py

```diff
@@ -24,15 +24,274 @@
     PreTrainedModel,
     StaticCache,
     T5ForConditionalGeneration,
+    VoxtralForConditionalGeneration,
     WhisperForConditionalGeneration,
 )
 from transformers.generation.configuration_utils import GenerationConfig
+from transformers.integrations.executorch import TorchExportableModuleForDecoderOnlyLM, sdpa_mask_without_vmap
+from transformers.masking_utils import AttentionMaskInterface
+from transformers.modeling_utils import AttentionInterface
 
 from optimum.executorch.attentions.custom_sdpa import get_custom_sdpa_for_ring_kv_cache
 
 from .utils import save_config_to_constant_methods
 
 
+class VoxtralEncoderExportableModule(torch.nn.Module):
+    """
+    Subgraph which handles all of the audio-related work: encoder, multimodal projection, combinining with text tokens.
+    The result of this subgraph should stream directly into the decoder subgraph.
+    """
+
+    def __init__(self, model: torch.nn.Module):
+        super().__init__()
+        self.audio_encoder = model.audio_tower
+        self.mm_projector = model.multi_modal_projector
+        self.intermediate_size = model.config.audio_config.intermediate_size
+        self.audio_token_id = model.config.audio_token_id
+        self.metadata = save_config_to_constant_methods(model.config.text_config, model.generation_config)
```

Comment:
is there anything about audio or vision config that need to be stored as well?

Reply:
Not that I know of, didn't need any of them to do the export